### PR TITLE
feat: add ghost user system

### PR DESCRIPTION
**Pull request by statebait**
## Ghost User System

Extension Changes - https://github.com/darkraichat/darkrai-extension/pull/6

When a user tries to login for the first time he is asked to pick a nickname (username). When he does this a register function will run, that will add the username to the database along with an auto generated password which is returned as a response and the extension will store both in the local storage. When ever a user uses the extension to login on to a websites chatroom, it should check if the user exists (from the credentials stored in the local storage) on the database and then check if the ghost password matches. If yes then it should send back a JWT which then will be used to establish the socket connection. If the user does not exist or the password is wrong then the user on the extension will be asked to pick a nickname (username) again and his existing credentials will be wiped from the chrome storage.

Further we could run a cron job that would check the last active time of a user and if it exceeds (a decided threshold) we purge that user. This is done to free up nicknames (usernames).

### TODO:
- [x] Add ghost user model
- [x] Add auto generating of ghost passwords
- [x] Add login using username/password
- [x] Add authentication with JWT to authenticate socket connections

### Some other time
- Write cron job to purge old users
- Update last active of a user when ever he sends a message

Other changes: 
adds cross-env to set NODE_ENV when starting the server
abstracted pulling initial messages to its own controller

Relates to #15

### CODE REVIEW

File: src/routes/index.js [hunk 1]
```diff
@@ -1,22 +1,70 @@
...
+
 const router = express.Router()
-const Message = require('../models/message')
 
-router.get('/login', (req, res) => {
-  const room = {
-    website: req.query.website,
-    hateSpeechFlag: false,
+router.post('/login', async (req, res) => {
+  const user = await User.findOne({
```
Comment (saifnsound) [inline src/routes/index.js:12]: Move the business logic of this route to a separate file, and break it into separate functions.

File: src/routes/index.js [hunk 2]
```diff
@@ -1,22 +1,70 @@
...
+      initialMessages,
+    })
+  } else {
+    res.status(400).send({ message: 'User not found' })
   }
-  Message.find(room, (err, messages) => {
+})
+
+router.post('/register', (req, res) => {
+  User.find({ username: req.body.username }, (err, user) => {
```
Comment (saifnsound) [inline src/routes/index.js:38]: Same as above.

File: src/models/user.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+const userSchema = new mongoose.Schema({
+  username: {
+    type: String,
+    required: true,
+    trim: true,
+    maxlength: 20,
+    unique: true,
+  },
+  password: {
+    type: String,
```
Comment (saifnsound) [inline src/models/user.js:12]: Specify min and max length

**Comment by statebait** [comment]
I think we should merge this because as per our discussion (offline) we will be restructuring the entire project and separating a lot of the logic into their own files.

**Pull request merged**